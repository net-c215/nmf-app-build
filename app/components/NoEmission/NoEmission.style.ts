import { StyleSheet } from "react-native";
import { screen } from "../../constants/Layout";
import layout from "../../style/layout";

export default StyleSheet.create({
  container: {
    ...layout.containerWithPadding
  },
  image: {
    width: "100%",
    height: screen.width - 100
  },
  imageView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  textView: {
    alignItems: "center",
    flex: 1
  },
  paragraph: {
    textAlign: "center",
    paddingVertical: 10
  },
  header: {
    paddingVertical: 20
  },
  button: {
    marginTop: 20
  }
});
