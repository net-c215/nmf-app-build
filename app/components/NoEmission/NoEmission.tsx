import React from "react";
import { View, Image } from "react-native";

import { Text } from "../";
import ImagesAssets from "../../constants/ImagesAssets";
import styles from "./NoEmission.style";
import { t } from "../../utils/translations";

interface Props {
  addEmission: () => void;
}

export default function NoEmission(props: Props) {
  return (
    <View style={styles.container}>
      <View style={styles.imageView}>
        <Image
          style={styles.image}
          resizeMode="contain"
          source={ImagesAssets.components.nmfStickerEarth}
        />
      </View>
      <View style={styles.textView}>
        <Text.H1 style={styles.header}>{t("NO_EMISSION_TITLE")}</Text.H1>
        <Text.Primary style={styles.paragraph}>
          Thanks for joining the fight to save our beautiful planet!
        </Text.Primary>
        <Text.Primary style={styles.paragraph}>
          You can start using the app by
          <Text.Primary bold> adding your first emission.</Text.Primary>
        </Text.Primary>
      </View>
    </View>
  );
}
