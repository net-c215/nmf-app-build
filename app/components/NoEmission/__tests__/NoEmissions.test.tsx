import React from "react";
import renderer from "react-test-renderer";

import NoEmission from "../";

jest.unmock("../");

it("renders correctly NoEmission", () => {
  const tree = renderer.create(<NoEmission />).toJSON();
  expect(tree).toMatchSnapshot();
});
