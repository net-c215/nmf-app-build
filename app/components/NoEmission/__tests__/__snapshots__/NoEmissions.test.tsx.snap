// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly NoEmission 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 650,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text.H1
      style={
        Object {
          "paddingVertical": 20,
        }
      }
    >
      Hi there ðŸ‘‹
    </Text.H1>
    <Text.Primary
      style={
        Object {
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Thanks for joining the fight to save our beautiful planet!
    </Text.Primary>
    <Text.Primary
      style={
        Object {
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      To start tracking your CO2 emissions, tap the button below
    </Text.Primary>
    <Button.Secondary
      onPress={[Function]}
      style={
        Object {
          "marginTop": 20,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        green={true}
      >
        Add my first emission
      </Text.Primary>
    </Button.Secondary>
  </View>
</View>
`;
